woocommerce-gutenberg-products-block.php

```diff
@@ -113,3 +113,39 @@ function() {
 
 add_action( 'plugins_loaded', array( '\Automattic\WooCommerce\Blocks\Package', 'init' ) );
 
+/**
+ * Gets JSON translation files from woocommerce core, if they exist, and appends them as a script.
+ *
+ * @param string $handle Script handle.
+ * @param string $relative_src Script filename/path.
+ */
+function woocommerce_blocks_add_woocommerce_core_translations( $handle, $relative_src ) {
```

Comment:
Are we okay with this running on every request (I don't think it's something we'll want to do)?  Can you think of a different approach that can avoid that?  

I wonder if using the [`pre_load_script_translations`](https://github.com/WordPress/wordpress-develop/blob/172f434332397bc9f5d7e35e9d3e27f8b66e864c/src/wp-includes/l10n.php#L1076) would be a better thing to hook in on?  That would also remove the need for adding a new `do_action` here?

Reply:
That hook doesn't give me the data I need. At least I need the handle and the filename...

Which part is worrying you? This code is largely pulled from WP core so regular script translation does exactly the same thing on every page load.

Reply:
> regular script translation does exactly the same thing on every page load.

Are you sure about this?  It looks to me like it only happens when a script is `enqueued`, not when it is `registered`.

Reply:
I will investigate this tomorrow. At first glance I think it may work the same, the only difference being I am using an inline script instead of leaving it for the script loader. Maybe it can be deferred. 

I could have another go at the other hook, but I would need a way to map handles to filenames. I also don't want it to short circuit regular translation loading. We need both sets of translations as new strings may be added to the plugin.